service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -147,6 +149,51 @@ public AssessmentQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTempla
     }
 
     @Override
+    public List<AssessmentInfo> findAssessmentInfoByKeys(final String clientName, final String... assessmentKeys) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("keys", Arrays.asList(assessmentKeys))
+            .addValue("clientName", clientName);
+
+        final String SQL =
+            "SELECT \n" +
+                "   a._key AS assessmentKey, \n" +
+                "   a.testid AS assessmentId, \n" +
+                "   tp.subjectname AS subject, \n" +
+                "   tp.label AS assessmentLabel, \n" +
+                "   GROUP_CONCAT(DISTINCT tool.code) languages, \n" +
+                "   GROUP_CONCAT(DISTINCT g.grade) grades, \n" +
+                "   tp.maxopportunities AS maxAttempts \n" +
+                "FROM \n" +
+                "   itembank.tblsetofadminsubjects a \n" +
+                "JOIN configs.client_testproperties tp \n" +
+                "   ON a.testid = tp.testid \n" +
+                "LEFT JOIN configs.client_testtool tool \n" +
+                "   ON tool.context = tp.testid \n" +
+                "LEFT JOIN itembank.setoftestgrades g \n" +
+                "   ON g.testid = tp.testid \n" +
+                "WHERE \n" +
+                "   tp.isselectable = 1 \n" +
+                "   AND tp.clientname = :clientName \n" +
+                "   AND tool.clientname = :clientName \n" +
+                "   AND tool.type = 'Language' \n" +
+                "   AND tool.contexttype = 'TEST' \n" +
+                "   AND a._key IN (:keys) \n" +
+                "GROUP BY \n" +
+                "   assessmentKey, assessmentId, subject, assessmentLabel, maxAttempts";
+
+        return jdbcTemplate.query(SQL, parameters, (rs, row) ->
+            new AssessmentInfo.Builder()
+                .withKey(rs.getString("assessmentKey"))
+                .withId(rs.getString("assessmentId"))
+                .withLabel(rs.getString("assessmentLabel"))
+                .withSubject(rs.getString("subject"))
+                .withMaxAttempts(rs.getInt("maxAttempts"))
+                .withLanguages(Arrays.asList(rs.getString("languages").split(",")))
+                .withGrades(Arrays.asList(rs.getString("grades").split(",")))
```

Comment:
In general I prefer using a Guava Splitter:
```
private static final Splitter COMMA_SPLITTER = Splitter.on(",");

.withGrades(COMMA_SPLITTER.splitToList(rs.getString("grades"));
```
Because it simplifies a lot of splitting logic.  But personal preference =)